site/content/troubleshooting-guide.md

```diff
@@ -1 +1,122 @@
-### TODO
\ No newline at end of file
+
+# Troubleshooting Guide
+
+This guide outlines some failure scenarios and their possible solutions.
+
+## Node.js not installed
+AWS.Deploy.Tools relies on AWS CDK to provision resources for the user's cloud application. AWS CDK requires Node.js to be installed in the user's machine.
+
+**Minimum Node.js version >= 10.13.0**
+
+See [here](https://nodejs.org/en/download/) to install Node.js on your system.
+
+## Docker related issues
+AWS.Deploy.Tool requires Docker to perform containerized deployments.
+
+The following recipes are container based:
+
+ - ASP.NET Core App to Amazon ECS using Fargate
+ - ASP.NET Core App to AWS App Runner
+ - Service on Amazon ECS using Fargate
+ - Scheduled Task on Amazon ECS using Fargate
+ - Push Container Images to Amazon Elastic Container Registry
+
+### Docker not installed
+See [here](https://docs.docker.com/get-docker/) to install Docker for your operating system.
+
+### Docker not running in Linux mode
+If you are on a Windows operating system, it is likely that you are running Docker in a Windows container. AWS.Deploy.Tools requires Docker to be running in a Linux container.
+
+See [here](https://docs.docker.com/desktop/windows/#switch-between-windows-and-linux-containers) to switch between Windows and Linux containers.
+
+### Failed to push Docker Image
+AWS.Deploy.Tools builds the Docker image and pushes it to Amazon Elastic Container Registry (Amazon ECR). 
+
+If a user is missing the required IAM permissions to perform actions against ECR repositories, the deployment may fail the following error message:
+
+```
+Failed to push Docker Image
+```
+
+See [here](https://docs.aws.amazon.com/AmazonECR/latest/userguide/repository-policy-examples.html) for guidance on how to set IAM policy statements to allow actions on Amazon ECR repositories.
+
+## Invalid project path provided
+
+The CLI deployment command takes in an optional `--project-path` option.
+
+For example:
+```
+dotnet aws deploy --project-path
```

Comment:
I think it'd be nice to show either an example or a placeholder in this block.

Like `dotnet` does [here](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build), maybe something like
```
dotnet aws deploy --project-path <PATH>
```

Reply:
Done